traveler-ios-ui/TravelerKitUI/ViewControllers/ErrorViewController.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  ErrorViewController.swift
+//  Traveler
+//
+//  Created by Ata Namvari on 2018-10-17.
+//  Copyright © 2018 Guestlogix Inc. All rights reserved.
+//
+
+import UIKit
+
+public protocol ErrorViewControllerDelegate: class {
+    func errorViewControllerDidRetry(_ controller: ErrorViewController)
+}
+
+open class ErrorViewController: UIViewController {
+    public weak var delegate: ErrorViewControllerDelegate?
+    
+    @IBOutlet weak var errorTitle: UILabel!
+    @IBOutlet weak var errorMessage: UILabel!
+    @IBOutlet weak var retryButton: UIButton!
+    @IBOutlet weak var errorMessageToErrorTitleVerticalConstraint: NSLayoutConstraint!
+    
+    var errorTitleString: String?
+    var errorMessageString: String?
+    var retryButtonString: String?
+    
+    public override func viewDidLoad() {
+        if delegate == nil {
+            retryButton.isHidden = true
+            retryButton.isEnabled = false
+            
+            if let retryButtonString = retryButtonString {
```

Comment:
Don't need this `if let`.

Reply:
Don't want to override default storyboard text unless retryButtonString is set

Reply:
Doesn't matter anyway, the if-let is inside the if that checks for the delegate to nil, which means that if the delegate does not exist you will hide the retry button. The consequence being that you're changing the title of a hidden button.